### DISCUSSION TITLE
What should I do after a failed patch for thunderbolt ?

### DISCUSSION THREAD

**Discussion by xiaooloong** [osy86/HaC-Mini#985] [Q&A]
Big Sur 11.6.4 (20G417) with 3.1.5

I have pathed and uninstalled the thunderbolt path many times to switch between macOS and Windows. And everything worked fine.

Just now I switch OS from Windows to macOS and I use the GUI patcher app to flash as usual. But it keep printing message and does not seem to finish. So I press cmd + q, quit the pathcher app and launch again.

This time, weather I flash nor uninstall it all failed with message `Exiting because existing data is not matching original.`

What should I do now?

**Comment by xiaooloong** [comment]
#270 seems the best way for me.

**Reply by xiaooloong** [reply]
After I received my ch341a, I dump the broken firmware and , yes, it is same as the dump file made by GUI patcher App.

Here is the broken firmware if anyone want to analyse.
[70-broken.zip](https://github.com/osy/HaC-Mini/files/8239153/70-broken.zip)

Instead of using the dump file made by GUI patcher App at first time before I pathed, I use the official firmware download from Intel website. It works perfectly on Windows now.

**Comment by osy** [comment]
For the record I do not recommend doing this as the SPI flash used by the controller does not have a high write endurance. You might end up bricking your SPI.
